pkg/cloudprovider/gcp.go

```diff
@@ -13,25 +19,171 @@ const (
+	for _, networkInterface := range instance.NetworkInterfaces {
+		keepAliases := []*google.AliasIpRange{}
+		for _, aliasIPRange := range networkInterface.AliasIpRanges {
+			if assignedIP := net.ParseIP(aliasIPRange.IpCidrRange); ip != nil && assignedIP != nil && !assignedIP.Equal(ip) {
+				keepAliases = append(keepAliases, aliasIPRange)
+				continue
+			}
+			if assignedIP, _, err := net.ParseCIDR(aliasIPRange.IpCidrRange); err == nil && !assignedIP.Equal(ip) {
+				keepAliases = append(keepAliases, aliasIPRange)
+			}
+		}
+		networkInterface.AliasIpRanges = keepAliases
+		operation, err := g.client.Instances.UpdateNetworkInterface(g.project, g.parseZone(instance.Zone), instance.Name, networkInterface.Name, networkInterface).Do()
+		if err != nil {
+			return nil, err
+		}
+		opName = operation.Name
+		break
+	}
+	return GCPWaitInput{
+		opName: opName,
+		zone:   g.parseZone(instance.Zone),
+	}, nil
 }
 
-func (a *GCP) WaitForResponse(interface{}) error {
-	return nil
+func (g *GCP) WaitForResponse(requestObj interface{}) error {
+	gcpWaitInput, ok := requestObj.(GCPWaitInput)
+	if !ok {
+		return fmt.Errorf("error decoding GCP requestObj, object not of type: GCPWaitInput %#v", requestObj)
+	}
+	_, err := g.client.ZoneOperations.Wait(g.project, gcpWaitInput.zone, gcpWaitInput.opName).Do()
+	return err
+}
+
+func (g *GCP) GetNodeSubnet(node *corev1.Node) (*net.IPNet, *net.IPNet, error) {
+	instance, err := g.getInstance(node)
+	if err != nil {
+		return nil, nil, err
+	}
+	var v4Subnet, v6Subnet *net.IPNet
+	for _, networkInterface := range instance.NetworkInterfaces {
+		region, subnet := g.parseSubnet(networkInterface.Subnetwork)
+		subnetResult, err := g.client.Subnetworks.Get(g.project, region, subnet).Do()
+		if err != nil {
+			return nil, nil, err
+		}
+		if subnetResult.IpCidrRange != "" {
+			_, v4Subnet, _ = net.ParseCIDR(subnetResult.IpCidrRange)
+		}
+		if subnetResult.Ipv6CidrRange != "" {
+			_, v6Subnet, _ = net.ParseCIDR(subnetResult.Ipv6CidrRange)
+		}
+		break
+	}
+	return v4Subnet, v6Subnet, nil
+}
+
+//  This is what the node's providerID looks like on GCP
+// 	spec:
+//   providerID: gce://openshift-gce-devel-ci/us-east1-b/ci-ln-pvr3lyb-f76d1-6w8mm-master-0
+//  i.e: projectID/zone/instanceName
+func (g *GCP) getInstance(node *corev1.Node) (*google.Instance, error) {
+	providerData := parseProviderID(node.Spec.ProviderID)
+	instance, err := g.client.Instances.Get(providerData[len(providerData)-3], providerData[len(providerData)-2], providerData[len(providerData)-1]).Do()
+	if err != nil {
+		return nil, err
+	}
+	return instance, nil
+}
+
+// GCP Subnet URLs are defined as:
+// - https://www.googleapis.com/compute/v1/projects/project/regions/region/subnetworks/subnetwork
+// OR
+// - regions/region/subnetworks/subnetwork
+func (g *GCP) parseSubnet(subnetURL string) (string, string) {
+	subnetURLParts := strings.Split(subnetURL, "/")
+	return subnetURLParts[len(subnetURLParts)-3], subnetURLParts[len(subnetURLParts)-1]
```

Comment:
Do a length check here, otherwise you'll panic.

Reply:
It might just be better if we panic if google decides to change their GCP API? At least like this it will be more "evident" to us